.github/workflows/workflow.yaml

```diff
@@ -0,0 +1,149 @@
+name: CI workflow
+
+on:
+  push: 
+    branches:
+      - 'release/v*'
+    tags:
+      - '*'
+    paths-ignore:
+      - '**.md'
+      - 'CODEOWNERS'
+      - 'LICENSE'
+      - 'docs/**'
+  pull_request:
+    branches:
+      - 'release/v*'
+    paths-ignore:
+      - '**.md'
+      - 'CODEOWNERS'
+      - 'LICENSE'
+      - 'docs/**'
+
+jobs:
+  ci:
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
```

Comment:
This job fails on the release/v1.4 branch due to the error "run out of disk space".  Although it does pass here, can we change it to run on the non-containerized runner to avoid the potential degradation?
```
 runs-on: runs-on,runner=4cpu-linux-x64,run-id=${{ github.run_id }}
```

Reply:
1.4 was running integration checks which have been moved to a separate job here. Have fixed it for 1.4 and it is passing there as well. Have added more details in the thread there https://github.com/rancher/rke/pull/3616#discussion_r1650635832
Have not faced this issue here or on 1.6 on multiple runs, so i belv it should be fine. I believe its better to stay on `ubuntu-latest` where possible since that is free.